build/build-treemap.js

```diff
@@ -0,0 +1,162 @@
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+* or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/`;
+
+/**
+ * Evaluates path glob and loads all identified files as an array of strings.
+ * @param {string} pattern
+ * @return {Promise<Array<string>>}
+ */
+async function loadFiles(pattern) {
+  const filePaths = await glob(pattern);
+  return Promise.all(filePaths.map(path => readFileAsync(path, {encoding: 'utf8'})));
+}
+
+/**
+ * Write a file to filePath, creating parent directories if needed.
+ * @param {string} filePath
+ * @param {string} data
+ * @return {Promise<void>}
+ */
+async function safeWriteFileAsync(filePath, data) {
+  const fileDir = path.dirname(filePath);
+  await mkdir(fileDir, {recursive: true});
+  return writeFileAsync(filePath, data);
+}
+
+/**
+ * Copy static assets.
+ * @return {Promise<void>}
+ */
+function copyAssets() {
+  return cpy([
+    'images/**/*',
+    'index.html',
+    'debug.json',
+  ], distDir, {
+    cwd: `${sourceDir}/app/`,
+    parents: true,
+  });
+}
+
+/**
+ * Concat report and viewer stylesheets into single treemap.css file.
+ * @return {Promise<void>}
+ */
+async function css() {
+  const viewerCss = await readFileAsync(`${sourceDir}/app/styles/treemap.css`, {encoding: 'utf8'});
+  await safeWriteFileAsync(`${distDir}/styles/treemap.css`, [viewerCss].join('\n'));
+}
+
+/**
+ * Insert report templates into html and copy to dist.
+ * @return {Promise<void>}
+ */
+async function html() {
+  const htmlSrc = await readFileAsync(`${sourceDir}/app/index.html`, {encoding: 'utf8'});
+  await safeWriteFileAsync(`${distDir}/index.html`, htmlSrc);
+}
+
+/**
+ * Combine multiple JS files into single treemap.js file.
+ * @return {Promise<void>}
+ */
+async function compileJs() {
+  // Treemap-specific JS files.
+  const srcJsFiles = await loadFiles(`${sourceDir}/app/src/*.js`);
+
+  const contents = [
+    `"use strict";`,
+    ...srcJsFiles,
+    fs.readFileSync(`${__dirname}/../node_modules/webtreemap-cdt/dist/webtreemap.js`, 'utf-8'),
+  ];
+
+  if (process.env.DEBUG) {
+    await safeWriteFileAsync(`${distDir}/src/treemap.js`, contents.join('\n'));
+    return;
+  }
+
+  const options = {
+    output: {preamble: license}, // Insert license at top.
+  };
+  const uglified = terser.minify(contents, options);
+  if (uglified.error || !uglified.code) {
+    throw uglified.error;
+  }
+
+  await safeWriteFileAsync(`${distDir}/src/treemap.js`, uglified.code);
+}
+
+/**
+ * Publish treemap to gh-pages branch.
+ * @return {Promise<void>}
+ */
+async function deploy() {
```

Comment:
feels odd that deployment is *part* of build, I would expect a separate step (especially if we can defer building to a standard tool :D)

Reply:
Same is true for viewer script.